src/Diff.re

```diff
@@ -36,8 +37,16 @@ module MakeDiff = (IO1: ImageIO.ImageIO, IO2: ImageIO.ImageIO) => {
     let diffOutput = outputDiffMask ? IO1.makeSameAsLayout(base) : base;
 
     let diffPixelQueue = Queue.create();
+    let diffLinesStack = diffLines ? Some(Stack.create()) : None;
+
     let countDifference = (x, y) => {
       diffPixelQueue |> Queue.push((x, y));
+
+      switch (diffLinesStack) {
+      | Some(stack) when stack |> Stack.is_empty => stack |> Stack.push(y)
+      | Some(stack) when stack |> Stack.top < y => stack |> Stack.push(y)
```

Comment:
Why is this check needed? Isn't the `y` coordinate always going in the positive direction?

Reply:
Yes but this is just protection from fools I think. But yeah it does not matter do the < or !=